# INT

In [10]:
Rohith_roll_no="18"
print (Rohith_roll_no)

18


In [12]:
print(type(Rohith_roll_no))
print(id(Rohith_roll_no))

<class 'str'>
1980907746480


In [1]:
x="300"
y="300"
print(id(x))
print(id(y))
# doubt - when i tried for no's after 256 for this also showing same location
# This is due to a feature known as "integer caching" or "small integer optimization" implemented in Python to optimize memory usage and improve performance.

2969156999088
2969156999088


In [23]:
x="-116"
y="-116"
print(id(x))
print(id(y))
# doubt - only working for -ve values but for values after 256 it's showing the same memory location
#integer caching optimization is limited to the range of -5 to 256

1980907905520
1980907908208


In [49]:
x=9
y=8

Add=x+y
difference = x-y
product=x*y
division=x/y
remainder=x%y
quotient=x//y
exponentiation=x**y

print("sum of the no's:", Add)
print("difference between two no's:", difference)
print("product of two no's:", product)
print("divison of two no's:", division)
print("remainder is :", remainder)
print("quotient of two no's:", quotient)
print("exponentiation:", exponentiation)

sum of the no's: 17
difference between two no's: 1
product of two no's: 72
divison of two no's: 1.125
remainder is : 1
quotient of two no's: 1
exponentiation: 43046721


In [53]:
x=9
y=8
print(x>y)
print(x<y)
print(x>=y)
print(x<=y)

True
False
True
False


In [54]:
x=9
y=8
print(x==y)
print(x!=y)

False
True


In [57]:
print(10 and 20)      
print(0 and 20)       
print(20 and 0)      
print(0 and 0)  
# and operator returns true if all the expressions are true,
# here 10 and 20 are non zero integeres so it will be true rest all are false

20
0
0
0


In [61]:
a=10
b=10
print(a is b)
print(a is not b)

a=1000
b=1000
print(a is b)
print(a is not b)
#The is and is not operators in Python are used to check if two variables refer to the same object in memory


True
False
False
True


In [62]:
print(10+(10*32)//2**5&20+(~(-10))<<2)

20


In [5]:
python= "python is easier than java"
print("is" in python)
print("or" in python)

True
False


In [9]:
#Bitwise
print(10&20)
print(10|20)
print(10^20)
print(~10)
print(10<<2)
print(10>>2)

0
30
30
-11
40
2


In [10]:
a = 0b1010000
print(a)

b = 0o7436
print(b)

c = 0xfade
print(c)

print(bin(80))

print(oct(3870))

print(hex(64222))

print(bin(0b1010000))

print(bin(0xfade))

print(oct(0xfade))

print(oct(0o7436))

print(hex(0b1010000))

print(hex(0xfade))

80
3870
64222
0b1010000
0o7436
0xfade
0b1010000
0b1111101011011110
0o175336
0o7436
0x50
0xfade


In [1]:
num= 12.8
print(num)
print(type(num))
print(id(num))


12.8
<class 'float'>
2845218926352


# Float

In [6]:
num_1=13.5
num_2=17.5
print(num_1>num_2)
print(num_1<num_2)
print(num_1>=num_2)
print(num_1<=num_2)


False
True
False
True


In [8]:
num_1=16.25
num_2=16.225
print(num_1==num_2)
print(num_1!=num_2)

False
True


In [9]:

print(10.20 and 20.30)    
print(0.0 and 20.30)        
print(20.30 and 0.0)        
print(0.0 and 0.0)         

print(10.20 or 20.30)        
print(0.0 or 20.30)         
print(20.30 or 0.0)         
print(0.0 or 0.0)           

print(not 10.20)           
print(not 0.0)             

20.3
0.0
0.0
0.0
10.2
20.3
20.3
0.0
False
True


In [12]:
a=10.5
b=10.5
print(id(a))
print(id(b))

2845246879632
2845246875984


In [11]:
a=10.5
b=10.5
a=b
print(id(a))
print(id(b))

2845246878320
2845246878320


# complex

In [13]:
a=10+20j
print(type(a))
print(id(a))

<class 'complex'>
2845246880176


In [14]:
a=10+20j
b=20+30j
Add= a+b
difference= a-b
product= a*b
divide= a/b
quotient= a**b
print("sum of num's :", Add)
print("difference of two num's :",difference)
print('division',divide)
print("quotient", quotient)


sum of num's : (30+50j)
difference of two num's : (-10-10j)
division (0.6153846153846154+0.0769230769230769j)
quotient (-2348247776258.4404+2822337984266.558j)


In [15]:
a=10+20j
b=20+30j
print(a==b)
print(a!=b)

False
True


In [16]:
print(10+20j and 20+30j)   
print(0+0j and 20+30j)      
print(20+30j and 0+0j)     
print(0+0j and 0+0j)        

print(10+20j or 20+30j)   
print(0+0j or 20+30j)      
print(20+30j or 0+0j)    
print(0+0j or 0+0j)        

print(not 10+20j)         
print(not 0+0j)       

(20+30j)
0j
0j
0j
(10+20j)
(20+30j)
(20+30j)
0j
False
True


In [17]:
a=10+20j
b=10+20j
print(a is b)
print(a is not b)

False
True


# Boolean

In [2]:
a= True
print(type(a))
print(id(a))

<class 'bool'>
140736605276368


In [4]:
a=34
b=123
print(id(a))
print(id(b))

2483019384080
2483019386928


Any non-zero value within the range 0 - 256 has same id

2483019384080
2483019386928

In [7]:
value1 = 33
value2 = 116

# Find sum of both values
sum_result = value1 + value2

# Find difference between them
difference_result = value1 - value2

# Find the product of both
product_result = value1 * value2

# Find value after dividing first value with second value
division_result = value1 / value2

# Find the remainder after dividing first value with second value
remainder_result = value1 % value2

# Find the quotient after dividing first value with second value
quotient_result = value1 // value2

# Find the result of first value to the power of second value
power_result = value1 ** value2

print("Sum:", sum_result)
print("Difference:", difference_result)
print("Product:", product_result)
print("Division:", division_result)
print("Remainder:", remainder_result)
print("Quotient:", quotient_result)
print("Power:", power_result)


Sum: 149
Difference: -83
Product: 3828
Division: 0.28448275862068967
Remainder: 33
Quotient: 0
Power: 140480817492157101714681650001417870816399396626767640507265025947582134754624289653211347072774383015726130402998042766126243204368705054673180340332597934179904752712113342081


In [8]:
value1 = True
value2 = False
print(value1>value2)
print(value1<value2)
print(value1>=value2)
print(value1<=value2)

True
False
True
False


In [10]:
value1 = True
value2 = False
print(value1==value2)
print(value1!=value2)

False
True


In [13]:
print(True and True)
print(False and True)
print(True and False)
print(False and False)
print()
print(True or True)
print(False or True)
print(True or False)
print(False or False)
print()
print(not True)
print(not False)

True
False
False
False

True
True
True
False

False
True


In [14]:
value1 = True
value2 = False

 #Bitwise and(&)
and_result = value1 & value2

   #Bitwise or(|)
or_result = value1 | value2

 #Bitwise XOR(^)
xor_result = value1 ^ value2

#Bitwise negation(~)
negation_result = ~value1

  #Bitwise left shift
left_shift_result = value1 << 2

##Bitwise right shift
right_shift_result = value1 >> 2

print("Bitwise AND:", and_result)
print("Bitwise OR:", or_result) 
print("Bitwise XOR:", xor_result)
print("Bitwise negation:", negation_result)
print("Bitwise left shift:", left_shift_result)
print("Bitwise right shift:", right_shift_result)


Bitwise AND: False
Bitwise OR: True
Bitwise XOR: True
Bitwise negation: -2
Bitwise left shift: 4
Bitwise right shift: 0


In [5]:
message = "Nitish birthday is coming soon, set a reminder for 15-June" #date= 15 month=june
res=message.split()
pirnt(res)

NameError: name 'pirnt' is not defined

In [8]:
a1 = [100, 50]
a2 = ["delhi", "mumbai"]

#############################
a2.extend(a1)
print(a2)

['delhi', 'mumbai', 100, 50]
